Extract msg keys from password validation rules

gabichelsea opened the pull request:
Trello card link:
[https://trello.com/c/CswWBSEV](https://trello.com/c/CswWBSEV)

gabichelsea committed:
Added @NoArgsConstructor annotation for UserDetailsAdapter class
Replaced constructor to @AllArgsConstructor annotation

gabichelsea committed:
Initial commit for 'Extract message keys from password validation rules' task

gabichelsea committed:
Fixed PasswordContainsWithUnCorrectParametersRuleTest
Added PasswordValidator interface again

gabichelsea committed:
Added enum with message keys

gabichelsea committed:
Now the PasswordValidator interface is the same as PasswordValidationRule.
I removed PasswordValidationRule interface

gabichelsea committed:
Added mapper

gabichelsea committed:
Added test cases for the PasswordValidationMessageKeyMapper class

gabichelsea committed:
Little refactoring

lamfalusy commented on service/src/main/java/com/epam/rft/atsy/service/passwordchange/validation/impl/PasswordOldPasswordMatchesRule.java:
```diff
@@ -3,44 +3,38 @@
       userDetails = authenticationService.getCurrentUserDetails();
     } catch (UserNotLoggedInException e) {
-      return false;
+      e.printStackTrace();
```
Use logger instead of printing to System.err.

lamfalusy commented on service/src/test/java/com/epam/rft/atsy/service/passwordchange/validation/impl/PasswordContainsWithCorrectParametersRuleTest.java:
```diff
@@ -0,0 +1,54 @@
+@RunWith(Parameterized.class)
+public class PasswordContainsWithCorrectParametersRuleTest {
+  private final String newPassword;
+  private final boolean expectedIsValid;
```
expectedIsValid field is private, but it is not used in this class.

gabichelsea replied:
The Parameterized test is used this variable.

lamfalusy commented on service/src/test/java/com/epam/rft/atsy/service/passwordchange/validation/impl/PasswordContainsWithUnCorrectParametersRuleTest.java:
```diff
@@ -13,11 +13,14 @@
 @RunWith(Parameterized.class)
 public class PasswordContainsWithUnCorrectParametersRuleTest {
   private final String newPassword;
+  private final boolean expectedIsValid;
```
This field (expectedIsValid) is not necessary.

gabichelsea replied:
The Parameterized test is used this variable.

lamfalusy commented on web/src/test/java/com/epam/rft/atsy/mapper/PasswordValidationMessageKeyMapperTest.java:
```diff
@@ -0,0 +1,66 @@
+package com.epam.rft.atsy.mapper;
```
This test class stray a little bit. The PasswordValidationMessageKeyMapper class is in com.epam.rft.atsy.web.mapper package, but this test class is in com.epam.rft.atsy.mapper package.

lamfalusy commented on web/src/main/java/com/epam/rft/atsy/web/validation/messagekey/PasswordValidationMessageKey.java:
```diff
@@ -0,0 +1,24 @@
+package com.epam.rft.atsy.web.validation.messagekey;
```
In my opinion this enum should be moved to com.epam.rft.atsy.web.mapper package, because the only class which use this enum is PasswordValidationRuleMapper, and if I know well, it is not necessary to use this enum elsewhere, so if this enum is moved to the PasswordValidationRuleMapper's package, then this enum's visibility modifier can be "package-private" (the default one).

gabichelsea committed:
Fixed it

gabichelsea committed:
Removed unused parameter

gabichelsea committed:
Removed @NoArgConstructor annotation from UserDetailsAdapter class

gabichelsea committed:
Merge branch 'master' into extract-msg-keys-from-password-validation-rules

# Conflicts:
#	web/src/main/java/com/epam/rft/atsy/web/controllers/PasswordChangeController.java

gabichelsea committed:
Merge remote-tracking branch 'remotes/origin/password-change-feature-fix' into extract-msg-keys-from-password-validation-rules

gabichelsea committed:
After merge, 1 test case is not working(PasswordChangeControllerTest.java(changePasswordShouldRespondWithModelAndViewContainingErrorMessageWhenPasswordValidationFails method)

gabichelsea committed:
Fixed this test method:
changePasswordShouldRespondWithModelAndViewContainingErrorMessageWhenPasswordValidationFails()

gabichelsea committed:
Removed (min = 1)

gabichelsea committed:
Merge branch 'fixed-email-field-in-candidate-dto' into extract-msg-keys-from-password-validation-rules
